resources/guides/laptop_guide.md

```diff
@@ -0,0 +1,72 @@
+---
+id: laptop_guide
+title: Laptop Requirements Guide
+sidebar_label: Laptop Requirements Guide
+slug: /guides/laptop
+---
+
+_Last updated on 2022/08/24_
+
+Here are a few things to keep in mind when purchasing/configuring your laptop:
+
+## Hardware
+
+Although the following specifications are not strictly required, having these minimum specifications will make your computer less frustrating to use:
+
+-   8 GB of RAM
+    -   for gaming/future-proof purposes, 12 or 16 GB is more preferable
+-   256 GB of SSD storage
+    -   UWindsor provides 1 TB OneDrive cloud storage. However, if you download a lot of games and software, or if you simply prefer more local storage, you should upgrade to 512 GB / 1 TB storage
+
+## Operating System
+
+You aren't required to have certain OS since the School of Computer Science has remote Linux server that you can connect to, but here are a few pros and cons to help you decide which OS to use:
+
+### Windows
+
+Windows is one of the most popular OS among students, particularly because it comes with most pre-assembled laptops. The user interface should be fairly intuitive even when you are new to using laptops, and it is one of the most supported OS for app availability or IT support. Additionally, it is much easier to game on Windows.
+
+Many Computer Science classes (COMP-1400 and COMP-1410, for example) will require programming environments that are harder to set up in Windows than in other UNIX/Linux-based OS. Although substitutes like WSL and virtual machines can help, they may not be fully compatible or require a lot of computing power. Should you choose Windows as your primary OS, you should consider setting up an SSH connection to the UWindsor's remote server on editors like Visual Studio Code or via other means. In addition, when you are working with other developers on a project, you may also have to pay attention to your line breaks (CRLF vs. LF).
```

Comment:
```suggestion
Many Computer Science classes (COMP-1400 and COMP-1410, for example) will require programming environments that are harder to set up in Windows than in other UNIX/Linux-based OS. Although substitutes like WSL (Windows Subsystem for Linux) and VMs (virtual machines) can help, they can require a lot of computing power (VMs especially). Should you choose Windows as your primary OS, you should consider setting up an SSH connection to the UWindsor's remote server on editors like Visual Studio Code or via other means. In addition, when you are working with other developers on a project, you may also have to pay attention to your line breaks (CRLF vs. LF).
```

Its not true that WSL or VMs have compatibility issues, at least insofar as coursework is concerned. In both cases, you're essentially running a guest Linux distro which behaves near-identical to a real system.

Reply:
Also, I believe we have an article for setting up WSL and connecting to CS servers with SSH in VS Code. We should link that here (and if we don't, we definitely need to create them)

Reply:
WSL does have some compatibility issues with some of the courses (256 and 330) however virtual machines should not have any compatibility issues.